src/libsyntax/errors/emitter.rs

```diff
@@ -361,16 +361,28 @@ fn print_diagnostic(dst: &mut Destination,
                     msg: &str,
                     code: Option<&str>)
                     -> io::Result<()> {
-    if !topic.is_empty() {
-        dst.start_attr(term::Attr::ForegroundColor(lvl.color()))?;
-        write!(dst, "{}: ", topic)?;
-        dst.reset_attrs()?;
-    }
 
-    dst.start_attr(term::Attr::Bold)?;
-    write!(dst, "{}: ", lvl.to_string())?;
-    write!(dst, "{}", msg)?;
-    dst.reset_attrs()?;
+
+    match lvl {
+        Level::Error => {
```

Comment:
Ah, good point, I missed that.  Right, the important piece is to handle the top line differently.  Do we want a separate header renderer?